diff --git a/toolkit/moz.configure b/toolkit/moz.configure
index e5e517635d639..4482baba59699 100644
--- a/toolkit/moz.configure
+++ b/toolkit/moz.configure
@@ -22,6 +22,7 @@ def check_moz_app_id(moz_app_id, build_project):

 project_flag(
     env="MOZ_APP_VENDOR",
+    default="Cachy",
     nargs=1,
     help='Used for application.ini\'s "Vendor" field, which also impacts profile location and user-visible fields',
 )
@@ -35,6 +36,7 @@ project_flag(

 project_flag(
     "MOZ_APP_PROFILE",
+    default="cachy",
     nargs=1,
     help='Used for application.ini\'s "Profile" field, which controls profile location',
 )
