diff -Naur a/security/moz.build b/security/moz.build
--- a/security/moz.build	2022-12-13 07:48:56.201367932 +0200
+++ b/security/moz.build	2022-12-14 09:22:55.687549525 +0200
@@ -59,11 +59,9 @@
     ]
     gyp_vars["nspr_libs"] = "nspr"
 else:
+    # Build mozpkix and mozpkix-test only
     gyp_vars["nspr_libs"] = "nspr"
-    if not CONFIG["ENABLE_TESTS"]:
-        # If we're not building tests, then we don't need a static copy
-        # of NSS. Build mozpkix only.
-        gyp_vars["mozpkix_only"] = 1
+    gyp_vars["mozpkix_only"] = 1
 
 # This disables building some NSS tools.
 gyp_vars["mozilla_client"] = 1
