--- a/browser/base/content/main-popupset.inc.xhtml
+++ b/browser/base/content/main-popupset.inc.xhtml
@@ -652,6 +652,8 @@
               checked="false"
               autocheck="false"
               oncommand="FullPageTranslationsPanel.onNeverTranslateSite()"/>
+    <menuitem data-l10n-id="translations-panel-disable-translations"
+              oncommand="FullPageTranslationsPanel.openDisableTranslations()"/>
     <menuseparator/>
     <menuitem class="manage-languages-menuitem"
               data-l10n-id="translations-panel-settings-manage-languages"
--- a/browser/components/preferences/main.inc.xhtml
+++ b/browser/components/preferences/main.inc.xhtml
@@ -381,27 +381,36 @@
           preference="layout.spellcheckDefault"/>
 
   <!-- Translations -->
-  <vbox id="translationsGroup" hidden="true" data-subcategory="translations">
+  <vbox id="translationsGroup" data-subcategory="translations">
     <label><html:h2 data-l10n-id="translations-manage-header"/></label>
-    <hbox id="translations-manage-description" align="center">
-      <description flex="1" data-l10n-id="translations-manage-intro"/>
-      <button id="translations-manage-settings-button"
-              is="highlightable-button"
-              class="accessory-button"
-              data-l10n-id="translations-manage-settings-button"/>
-    </hbox>
-    <vbox>
-      <html:div id="translations-manage-install-list" hidden="true">
-        <hbox class="translations-manage-language">
-          <label data-l10n-id="translations-manage-install-description"></label>
-          <button id="translations-manage-install-all"
-                  data-l10n-id="translations-manage-language-install-all-button"></button>
-          <button id="translations-manage-delete-all"
-                  data-l10n-id="translations-manage-language-remove-all-button"></button>
-        </hbox>
-        <!-- The downloadable languages will be listed here. -->
-      </html:div>
-      <description id="translations-manage-error" hidden="true"></description>
+    <checkbox id="translations-manage-enable" 
+              data-l10n-id="translations-manage-enable" 
+              preference="browser.translations.enable"
+              data-subcategory="translations-enable" />
+    <vbox id="innerTranslationsGroup" hidden="true">
+      <checkbox id="translations-manage-autopopup" 
+                data-l10n-id="translations-manage-autopopup" 
+                preference="browser.translations.automaticallyPopup" />
+      <hbox id="translations-manage-description" align="center">
+        <description flex="1" data-l10n-id="translations-manage-intro"/>
+        <button id="translations-manage-settings-button"
+                is="highlightable-button"
+                class="accessory-button"
+                data-l10n-id="translations-manage-settings-button"/>
+      </hbox>
+      <vbox>
+        <html:div id="translations-manage-install-list" hidden="true">
+          <hbox class="translations-manage-language">
+            <label data-l10n-id="translations-manage-install-description"></label>
+            <button id="translations-manage-install-all"
+                    data-l10n-id="translations-manage-language-install-all-button"></button>
+            <button id="translations-manage-delete-all"
+                    data-l10n-id="translations-manage-language-remove-all-button"></button>
+          </hbox>
+          <!-- The downloadable languages will be listed here. -->
+        </html:div>
+        <description id="translations-manage-error" hidden="true"></description>
+      </vbox>
     </vbox>
   </vbox>
 </groupbox>
--- a/browser/components/preferences/main.js
+++ b/browser/components/preferences/main.js
@@ -144,6 +144,10 @@ Preferences.addAll([
   { id: "dom.ipc.processCount.web", type: "int" },
   { id: "layers.acceleration.disabled", type: "bool", inverted: true },
 
+  /* Fullpage Translations */
+  { id: "browser.translations.enable", type: "bool" },
+  { id: "browser.translations.automaticallyPopup", type: "bool" },
+
   // Files and Applications
   { id: "pref.downloads.disable_button.edit_actions", type: "bool" },
 
@@ -325,6 +329,22 @@ var gMainPane = {
 
     gMainPane.initTranslations();
 
+    let inPrompt = false;
+    Preferences.get("browser.translations.enable").on("change", () => {
+      if(!Preferences.get("browser.translations.enable").value)
+      if(!inPrompt) {
+        inPrompt = true;
+        confirmRestartPrompt(false, 1, true, false).then(buttonIndex => {
+          inPrompt = false;
+          if (buttonIndex == CONFIRM_RESTART_PROMPT_RESTART_NOW) {
+            Services.startup.quit(
+              Ci.nsIAppStartup.eAttemptQuit | Ci.nsIAppStartup.eRestart
+            );
+          }
+        });
+      }
+    });
+
     if (
       Services.prefs.getBoolPref(
         "media.videocontrols.picture-in-picture.enabled"
@@ -771,6 +791,10 @@ var gMainPane = {
       document.getElementById("checkSpelling"),
       () => this.writeCheckSpelling()
     );
+    Preferences.addSyncFromPrefListener(
+      document.getElementById("translations-manage-enable"),
+      () => this.readEnableTranslations()
+    );
     Preferences.addSyncFromPrefListener(
       document.getElementById("alwaysAsk"),
       () => this.readUseDownloadDir()
@@ -1001,24 +1025,33 @@ var gMainPane = {
     document.getElementById("zoomBox").hidden = false;
   },
 
+  readEnableTranslations(skipInit = false) {
+    const translationsEnabled = Preferences.get("browser.translations.enable").value;
+    document.getElementById("innerTranslationsGroup").hidden = !translationsEnabled;
+    if (!this._translationsInitialized && !skipInit)
+      this.initTranslations();
+  },
+
+  _translationsInitialized: false,
+
   /**
    * Initialize the translations view.
    */
   async initTranslations() {
+    this.readEnableTranslations(true);
+
     if (!Services.prefs.getBoolPref("browser.translations.enable")) {
       return;
     }
 
+    this._translationsInitialized = true;
+
     /**
      * Which phase a language download is in.
      *
      * @typedef {"downloaded" | "loading" | "uninstalled"} DownloadPhase
      */
 
-    // Immediately show the group so that the async load of the component does
-    // not cause the layout to jump. The group will be empty initially.
-    document.getElementById("translationsGroup").hidden = false;
-
     class TranslationsState {
       /**
        * The fully initialized state.
--- a/browser/components/translations/content/fullPageTranslationsPanel.js
+++ b/browser/components/translations/content/fullPageTranslationsPanel.js
@@ -1239,6 +1239,15 @@ var FullPageTranslationsPanel = new (class {
     );
   }
 
+  openDisableTranslations() {
+    const window =
+      gBrowser.selectedBrowser.browsingContext.top.embedderElement.ownerGlobal;
+    window.openTrustedLinkIn(
+      "about:preferences#general-translations-enable",
+      "tab"
+    );
+  }
+
   /**
    * Redirect the user to about:preferences
    */
--- a/browser/locales/en-US/browser/translations.ftl
+++ b/browser/locales/en-US/browser/translations.ftl
@@ -58,6 +58,9 @@ translations-panel-settings-never-translate-unknown-language =
 translations-panel-settings-never-translate-site =
   .label = Never translate this site
 
+translations-panel-disable-translations =
+  .label = Disable translations completely
+
 ## The translation panel appears from the url bar, and this view is the default
 ## translation view.
 
@@ -123,6 +126,10 @@ translations-manage-header = Translations
 translations-manage-settings-button =
     .label = Settings…
     .accesskey = t
+translations-manage-enable = 
+  .label = Enable fullpage translations
+translations-manage-autopopup = 
+  .label = Prompt to translate pages that aren’t in the browser’s configured language
 translations-manage-intro = Set your language and site translation preferences and manage languages installed for offline translation.
 translations-manage-install-description = Install languages for offline translation
 translations-manage-language-install-button =
