--- a/browser/components/customizableui/CustomizableUI.jsm
+++ b/browser/components/customizableui/CustomizableUI.jsm
@@ -277,7 +277,6 @@ var CustomizableUIInternal = {
       {
         type: CustomizableUI.TYPE_TOOLBAR,
         defaultPlacements: [
-          "firefox-view-button",
           "tabbrowser-tabs",
           "new-tab-button",
           "alltabs-button",
@@ -605,18 +604,6 @@ var CustomizableUIInternal = {
         navbarPlacements.splice(newPosition, 0, "save-to-pocket-button");
       }
     }
-
-    // Add firefox-view if not present
-    if (currentVersion < 18) {
-      let tabstripPlacements =
-        gSavedState.placements[CustomizableUI.AREA_TABSTRIP];
-      if (
-        tabstripPlacements &&
-        !tabstripPlacements.includes("firefox-view-button")
-      ) {
-        tabstripPlacements.unshift("firefox-view-button");
-      }
-    }
   },
 
   _updateForNewProtonVersion() {
--- a/browser/components/firefoxview/firefoxview.html
+++ b/browser/components/firefoxview/firefoxview.html
@@ -113,7 +113,7 @@
         </div>
       </template>
 
-      <details class="content-container" is="tab-pickup-container" id="tab-pickup-container" open>
+      <details class="content-container" is="tab-pickup-container" id="tab-pickup-container" open hidden="true">
         <summary class="page-section-header">
           <h1 id="collapsible-synced-tabs-header" data-l10n-id="firefoxview-tabpickup-header"></h1>
           <span class="twisty icon" data-l10n-id="firefoxview-collapse-button-hide" aria-role="presentation"></span>
@@ -141,7 +141,7 @@
         </div>
       </details>
 
-      <aside id="colorway-landmark" class="content-container" is="colorways-card">
+      <aside id="colorway-landmark" class="content-container" is="colorways-card" hidden="true">
         <template id="colorways-no-collection-template">
           <div class="content-container">
             <div id="no-current-colorway-collection-notice" data-l10n-id="firefoxview-no-current-colorway-collection"></div>
--- a/browser/components/firefoxview/firefoxview.mjs
+++ b/browser/components/firefoxview/firefoxview.mjs
@@ -28,7 +28,26 @@ const MediaQueryDOMSorting = {
   },
 };
 
+function onPrefChange() {
+  document.getElementById("tab-pickup-container").hidden =
+    Services.prefs.getBoolPref("identity.fxaccounts.enabled") &&
+    !Services.prefs.getBoolPref("cachy-browser.firefoxView.hideTabPickup", false)
+      ? false
+      : true;
+  document.getElementById("colorway-landmark").hidden =
+    !Services.prefs.getBoolPref("cachy-browser.firefoxView.hideColorways", true)
+      ? false
+      : true;
+}
+
+[
+  "cachy-browser.firefoxView.hideTabPickup",
+  "cachy-browser.firefoxView.hideColorways",
+  "identity.fxaccounts.enabled",
+].forEach((pref) => Services.prefs.addObserver(pref, onPrefChange));
+
 window.addEventListener("DOMContentLoaded", async () => {
+  onPrefChange();
   Services.telemetry.setEventRecordingEnabled("firefoxview", true);
   Services.telemetry.recordEvent("firefoxview", "entered", "firefoxview", null);
   document.getElementById("recently-closed-tabs-container").onLoad();
--- a/browser/base/content/navigator-toolbox.inc.xhtml
+++ b/browser/base/content/navigator-toolbox.inc.xhtml
@@ -41,16 +41,6 @@
       <hbox flex="1" align="end" class="toolbar-items">
         <toolbartabstop/>
         <hbox id="TabsToolbar-customization-target" flex="1">
-          <toolbarbutton id="firefox-view-button"
-                         class="toolbarbutton-1 chromeclass-toolbar-additional"
-                         data-l10n-id="toolbar-button-firefox-view"
-                         role="button"
-                         aria-pressed="false"
-                         oncommand="FirefoxViewHandler.openTab(event);"
-                         onmousedown="FirefoxViewHandler.openTab(event);"
-                         cui-areatype="toolbar"
-                         removable="true"/>
-
           <tabs id="tabbrowser-tabs"
                 is="tabbrowser-tabs"
                 flex="1"
@@ -689,6 +679,16 @@
                    closemenu="none"
                    cui-areatype="toolbar"
                    data-l10n-id="navbar-library"/>
+
+    <toolbarbutton id="firefox-view-button"
+                   class="toolbarbutton-1 chromeclass-toolbar-additional"
+                   data-l10n-id="toolbar-button-firefox-view"
+                   role="button"
+                   aria-pressed="false"
+                   oncommand="FirefoxViewHandler.openTab(event);"
+                   onmousedown="FirefoxViewHandler.openTab(event);"
+                   cui-areatype="toolbar"
+                   removable="true"/>
   </html:template>
 </toolbox>
 </box>
